### PR TITLE
Add compiled optimizer test for nadam

### PR DESCRIPTION
**Pull request by mlazos**
Fixes #ISSUE_NUMBER

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @peterbell10 @ipiszy @ngimel @yf225 @muchulee8

### CODE REVIEW

File: test/inductor/test_compiled_optimizers.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def tearDown(self):
     test_adamw = make_test(AdamW, lr=0.01)
     # Need to an impl which does not use python scalars
     # test_adamax = make_test(Adamax, lr=0.01)
-    # test_nadam = make_test(NAdam, lr=0.01)
+    test_nadam = make_test(NAdam, lr=0.01)
```
Comment (janeyx99) [inline test/inductor/test_compiled_optimizers.py:166]: can you add more configs as in https://github.com/pytorch/pytorch/blob/main/test/optim/test_optim.py#L864-L879

Comment (mlazos) [reply test/inductor/test_compiled_optimizers.py:166]: I don't think that's for this PR, I was merely adding something that I already knew we supported (nadam)

I can add additional tests for those too, I'm not sure if lr as a tensor works atm, I'll have to try it with all of them.

**Review by janeyx99 — APPROVED** [review]

**Comment by janeyx99** [comment]
Approving cuz adding tests is always better! Would agree + vouch for adding more tests once support is more solid :D

**Pull request closed without merging**